parser.go

```diff
@@ -651,6 +651,30 @@ func (p *parseState) addArgs(args ...string) error {
 	for len(p.positional) > 0 && len(args) > 0 {
 		arg := p.positional[0]
 
+		if !arg.isRemaining() && len(arg.Choices) != 0 {
+			found := false
+
+			for _, choice := range arg.Choices {
```

Comment:
Want to use `slices.Contains` here? or are you trying to preserve compatibility with older Go versions?

Reply:
Compatibility - upstream still requires >= 1.20 and I might try to get this taken upstream at some point, if there's any appetite for it.